Sources/NIOHTTP1/HTTPDecoder.swift

```diff
@@ -185,15 +187,23 @@ public class HTTPDecoder<HTTPMessageT>: ByteToMessageDecoder, AnyHTTPDecoder {
 
     private var parser = http_parser()
     private var settings = http_parser_settings()
+    private var decoding: Bool = false
 
-    fileprivate var pendingInOut: [NIOAny] = []
     fileprivate var state = HTTPParserState()
 
     fileprivate init(type: HTTPMessageT.Type) {
         /* this is a private init, the public versions only allow HTTPClientResponsePart and HTTPServerRequestPart */
         assert(HTTPMessageT.self == HTTPClientResponsePart.self || HTTPMessageT.self == HTTPServerRequestPart.self)
     }
 
+    deinit {
+        // Remove the stored reference to ChannelHandlerContext
+        self.parser.data = UnsafeMutableRawPointer(bitPattern: 0x0000deadbeef0000)
```

Comment:
to make @helje5's life easier and not cause needless merge conflicts, could you make this value `0xdeadbeef`? That way it'll work on 32-bit platforms too